packages/node/src/index.ts

```diff
@@ -308,7 +308,7 @@ async function compile(
       !file.endsWith('.mjs') &&
       !file.match(libPathRegEx)
   );
-  if (esmPaths.length) {
+  if (!isEdgeFunction && esmPaths.length) {
```

Comment:
Lets try this behind a flag?

```suggestion
  const shouldCompile = !isEdgeFunction || (isEdgeFunction && process.env.VERCEL_SKIP_COMPILE_EDGE_FUNC === '1')
  if (shouldCompile&& esmPaths.length) {
```